tests/core/actions/test_forms.py

```diff
@@ -1334,3 +1359,267 @@ async def test_ask_for_slot_if_not_utter_ask(
+            "from_intent",
+        ),
+        (
+            # `ignored_intents` as a list of 2 values and slot's not_intent has one
+            # value different than the ones in `ignored_intents`.
+            ["chitchat", "greet"],
+            ["inform"],
+            "from_intent",
+        ),
+        (
+            # `ignored_intents` as a list of 2 values and slot's not_intent has one
+            # value that is included also in `ignored_intents`.
+            ["chitchat", "greet"],
+            ["chitchat"],
+            "from_intent",
+        ),
+        # same examples for entity_type -> from_trigger_intent
+        (
+            # `ignored_intents` as a string and slot's not_intent as an empty list.
+            "greet",
+            [],
+            "from_trigger_intent",
+        ),
+        (
+            # `ignored_intents` as an empty list and slot's not_intent has a value.
+            [],
+            ["greet"],
+            "from_trigger_intent",
+        ),
+        (
+            # `ignored_intents` as a list of 2 values and slot's not_intent has one
+            # value different than the ones in `ignored_intents`.
+            ["chitchat", "greet"],
+            ["inform"],
+            "from_trigger_intent",
+        ),
+        (
+            # `ignored_intents` as a list of 2 values and slot's not_intent has one
+            # value that is included also in `ignored_intents`.
+            ["chitchat", "greet"],
+            ["chitchat"],
+            "from_trigger_intent",
+        ),
+    ],
+)
+def test_ignored_intents_with_other_type_of_slots(
+    ignored_intents: Union[Text, List[Text]],
+    slot_not_intent: Union[Text, List[Text]],
+    type: Text,
+):
+    form_name = "some_form"
+    entity_name = "some_slot"
+    form = FormAction(form_name, None)
+
+    domain = Domain.from_dict(
+        {
+            "forms": {
+                form_name: {
+                    IGNORED_INTENTS: ignored_intents,
+                    REQUIRED_SLOTS_KEY: {
+                        entity_name: [
+                            {
+                                "type": type,
+                                "value": "affirm",
+                                "intent": "true",
+                                "not_intent": slot_not_intent,
+                            }
+                        ],
+                    },
+                }
+            }
+        }
+    )
+
+    tracker = DialogueStateTracker.from_events(
+        "default",
+        [
+            SlotSet(REQUESTED_SLOT, "some_slot"),
+            UserUttered(
+                "hello",
+                intent={"name": "greet", "confidence": 1.0},
+                entities=[{"entity": entity_name, "value": "some_value"}],
+            ),
+            ActionExecuted(ACTION_LISTEN_NAME),
+        ],
+    )
+
+    slot_values = form.extract_other_slots(tracker, domain)
+    assert slot_values == {}
+
+
+def test_form_with_no_required_slots_keyword():
```

Comment:
can we please move this test to `test_domain`? It doesn't execute any code within forms but purely tests implementation from the `domain` module. 
Can you please also test that this gives a `FutureWarning`?